course/flow.py

```diff
@@ -1017,10 +1017,11 @@ def expire_flow_session(
 
     assert isinstance(grading_rule, FlowSessionGradingRule)
 
-    if (past_due_only
-            and grading_rule.due is not None
-            and now_datetime < grading_rule.due):
-        return False
+    if past_due_only:
+        if grading_rule.due is None:
+            return False
+        elif now_datetime < grading_rule.due:
+            return False
```

Comment:
Do you think if due is None, then never due is a better behavior?

Reply:
I think that's what that implements, no?

Reply:
No,  when doing bulk expire in progress session.

Reply:
Oh, whoops. Half-asleep apparently. I looked at the code with the green background. :)

Yeah, that seems fine semantically.

Reply:
: -). NVM.